app/src/main/kotlin/com/simplemobiletools/flashlight/screens/MainScreen.kt

```diff
@@ -0,0 +1,245 @@
+import com.simplemobiletools.commons.compose.extensions.AdjustNavigationBarColors
+import com.simplemobiletools.commons.compose.extensions.MyDevices
+import com.simplemobiletools.commons.compose.menus.ActionItem
+import com.simplemobiletools.commons.compose.menus.ActionMenu
+import com.simplemobiletools.commons.compose.menus.OverflowMode
+import com.simplemobiletools.commons.compose.settings.scaffold.SettingsLazyScaffold
+import com.simplemobiletools.commons.compose.settings.scaffold.topAppBarColors
+import com.simplemobiletools.commons.compose.settings.scaffold.topAppBarInsets
+import com.simplemobiletools.commons.compose.settings.scaffold.topAppBarPaddings
+import com.simplemobiletools.commons.compose.theme.AppThemeSurface
+import com.simplemobiletools.flashlight.R
+import com.simplemobiletools.flashlight.views.SleepTimer
+import kotlinx.collections.immutable.toImmutableList
+
+@Composable
+internal fun MainScreen(
+    timerText: String,
+    timerVisible: Boolean,
+    onTimerClosePress: () -> Unit,
+    flashlightActive: Boolean,
+    onFlashlightPress: () -> Unit,
+    showBrightDisplayButton: Boolean,
+    onBrightDisplayPress: () -> Unit,
+    showSosButton: Boolean,
+    sosActive: Boolean,
+    onSosButtonPress: () -> Unit,
+    showStroboscopeButton: Boolean,
+    stroboscopeActive: Boolean,
+    onStroboscopeButtonPress: () -> Unit,
+    showBrightnessBar: Boolean,
+    brightnessBarValue: Float,
+    onBrightnessBarValueChange: (Float) -> Unit,
+    showStroboscopeBar: Boolean,
+    stroboscopeBarValue: Float,
+    onStroboscopeBarValueChange: (Float) -> Unit,
+    showMoreApps: Boolean,
+    openSettings: () -> Unit,
+    openAbout: () -> Unit,
+    openSleepTimer: () -> Unit,
+    moreAppsFromUs: () -> Unit,
+) {
+    AdjustNavigationBarColors()
+    SettingsLazyScaffold(
+        customTopBar = { scrolledColor: Color, _: MutableInteractionSource, scrollBehavior: TopAppBarScrollBehavior, statusBarColor: Int, colorTransitionFraction: Float, contrastColor: Color ->
+            TopAppBar(
+                title = {},
+                actions = {
+                    val actionMenus = remember {
+                        val settings =
+                            ActionItem(R.string.settings, icon = Icons.Filled.Settings, doAction = openSettings, overflowMode = OverflowMode.NEVER_OVERFLOW)
+                        val about = ActionItem(R.string.about, icon = Icons.Outlined.Info, doAction = openAbout, overflowMode = OverflowMode.NEVER_OVERFLOW)
+                        val sleepTimer = ActionItem(R.string.sleep_timer, doAction = openSleepTimer, overflowMode = OverflowMode.ALWAYS_OVERFLOW)
+                        val list = mutableListOf(settings, about, sleepTimer)
+                        if (showMoreApps) {
+                            list += ActionItem(R.string.more_apps_from_us, doAction = moreAppsFromUs, overflowMode = OverflowMode.ALWAYS_OVERFLOW)
+                        }
+                        list.toImmutableList()
```

Comment:
this can be a function no?

Reply:
Do you mean like a function that is passed to this composable?

Reply:
a separate function that creates the list as immutable, lots of logic in one place